comicapi/metadata/comet.py

```diff
@@ -0,0 +1,315 @@
+        return False
+
+    def name(self) -> str:
+        return "Comic Metadata (CoMet)"
+
+    @classmethod
+    def _get_parseable_credits(cls) -> list[str]:
+        parsable_credits: list[str] = []
+        parsable_credits.extend(cls._writer_synonyms)
+        parsable_credits.extend(cls._penciller_synonyms)
+        parsable_credits.extend(cls._inker_synonyms)
+        parsable_credits.extend(cls._colorist_synonyms)
+        parsable_credits.extend(cls._letterer_synonyms)
+        parsable_credits.extend(cls._cover_synonyms)
+        parsable_credits.extend(cls._editor_synonyms)
+        return parsable_credits
+
+    def _metadata_from_bytes(self, string: bytes, archive: Archiver) -> GenericMetadata:
+        tree = ET.ElementTree(ET.fromstring(string))
+        return self._convert_xml_to_metadata(tree, archive)
+
+    def _bytes_from_metadata(self, metadata: GenericMetadata) -> bytes:
+        tree = self._convert_metadata_to_xml(metadata)
+        return ET.tostring(tree.getroot(), encoding="utf-8", xml_declaration=True)
+
+    def _convert_metadata_to_xml(self, metadata: GenericMetadata) -> ET.ElementTree:
+        # shorthand for the metadata
+        md = metadata
+
+        # build a tree structure
+        root = ET.Element("comet")
+        root.attrib["xmlns:comet"] = "http://www.denvog.com/comet/"
+        root.attrib["xmlns:xsi"] = "http://www.w3.org/2001/XMLSchema-instance"
+        root.attrib["xsi:schemaLocation"] = "http://www.denvog.com http://www.denvog.com/comet/comet.xsd"
+
+        # helper func
+        def assign(comet_entry: str, md_entry: Any) -> None:
+            if md_entry is not None:
+                ET.SubElement(root, comet_entry).text = str(md_entry)
+
+        # title is manditory
+        assign("title", md.title or "")
+        assign("series", md.series)
+        assign("issue", md.issue)  # must be int??
+        assign("volume", md.volume)
+        assign("description", md.description)
+        assign("publisher", md.publisher)
+        assign("pages", md.page_count)
+        assign("format", md.format)
+        assign("language", md.language)
+        assign("rating", md.maturity_rating)
+        assign("price", md.price)
+        assign("isVersionOf", md.is_version_of)
+        assign("rights", md.rights)
+        assign("identifier", md.identifier)
+        assign("lastMark", md.last_mark)
+        assign("genre", ",".join(md.genres))  # TODO repeatable
+
+        for c in md.characters:
+            assign("character", c.strip())
+
+        if md.manga is not None and md.manga == "YesAndRightToLeft":
+            assign("readingDirection", "rtl")
+
+        if md.year is not None:
+            date_str = f"{md.year:04}"
+            if md.month is not None:
+                date_str += f"-{md.month:02}"
+            assign("date", date_str)
+
+        page = md.get_cover_page_index_list()[0]
+        assign("coverImage", md.pages[page]["filename"])
```

Comment:
`md.pages` is empty then using the CLI `-t comet` but is filled when using the GUI.

```
/comictagger/comicapi/metadata/comet.py", line 198, in _convert_metadata_to_xml
    assign("coverImage", md.pages[page]["filename"])
IndexError: list index out of range
```

Reply:
What did you run to get this to happen?

Reply:
`-v -s -t comet -o -i -f "Batman: The Brave and the Bold #1.zip"`

Reply:
Auto-tag does the same in the GUI. I presume the pages aren't read when auto-tagging?

Reply:
I don't think you updated the branch
```plain
git fetch https://github.com/lordwelch/comictagger +metadata-plugin:lordwelch/metadata-plugin
git checkout lordwelch/metadata-plugin
```

Reply:
I did it again and the same result. I tried with a 2000AD too. You can't replicated?

Reply:
I figured it out it's the `--clear-metadata` option, somehow it got set to default to on a while ago when it shouldn't have and I put it back in bb67ab009e69aab9a7b23ba9df0beff4f4c751a5 and I usually only test with the default options

Reply:
It is fixed now. I'll probably merge this PR in tomorrow